File: Two-Sum.cpp
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4        for(int i=0 ; i<nums.size() ; i++){
5            for(int j=i+1; j<nums.size(); j++){
6                if(nums[i]+nums[j] == target){
7                    return {i,j};
8                }
9            }
10        }
11        return {};
12    }
13};
14
15/* using map for optimisation
16
17class Solution {
18public:
19    vector<int> twoSum(vector<int>& nums, int target) {
20        unordered_map<int, int>m;
21        int i = 0;
22for (int x : nums) {
23    int y = target - x;
24    if (m.find(y) != m.end()) {
25        return {m[y], i};
26    }
27    m[x] = i;
28    i++;
29}
30        return {};
31    }
32};
33*/

File: Two-Sum.java
1class Solution {
2    public int[] twoSum(int[] nums, int target) {
3        int n = nums.length;
4        HashMap<Integer, Integer> map = new HashMap<>();
5        for(int i=0; i<n; i++){
6            int comp = target - nums[i];
7            if(map.containsKey(comp)){
8                return new int[] {map.get(comp), i};
9            }
10            map.put(nums[i], i);
11        }
12        return new int[]{};
13    }
14}